### ISSUE TITLE
Fail fast deactivation

### ISSUE THREAD

**Issue by aicioara** [sindresorhus/grunt-concurrent#67]
I have a CPU intensive task (task1) and a Network intensive task (task2), which I want to run in parallel. Now, if task1 fails, task2 is killed. The issue is that task2 has to clean after itself and it does not get the chance.

Is there an option to: 
- Either let all tasks fully complete their job (preferred and easier)
- Or have some sort of `finally` clause

I am happy to try open a pull request if any of the choices are part of the philosophy of the library.

**Comment by sindresorhus** [comment]
I assume you want https://github.com/sindresorhus/grunt-concurrent/blob/54a87d4313ce481757fa92d3b7d56dcecf16d77c/tasks/concurrent.js#L62 changed from `SIGKILL` to `SIGTERM`?

**Comment by aicioara** [comment]
I believe so, since a terminal signal kills it gracefully. Let me clone and test this and get back to you.

**Comment by kevva** [comment]
@aicioara, ping :).

**Comment by aicioara** [comment]
Sorry about the delay. 

I started playing with it. Replacing `SIGKILL` with `SIGTERM` has no effect. On further investigation, I discovered that

```javascript
// make sure all child processes are killed when grunt exits
process.on('exit', function () {
	grunt.log.writeln("This never gets called");
	cpCache.forEach(function (el) {
		el.kill('SIGKILL');
	});
});
```

Grunt never logs my message. My testcase consists of 1 long task (running all tests) and 1 short task (a linter). The linter is failing, my tests are interrupted, but the message in the console is not printed.
